control-plane-operator/controllers/hostedcontrolplane/v2/autoscaler/deployment.go

```diff
@@ -11,34 +13,189 @@ import (
+	AzureNodepoolLabel = "kubernetes.azure.com/agentpool"
+)
+
+func (a AutoscalerArg) String() string {
+	return string(a)
+}
 
-		if options.MaxPodGracePeriod != nil {
-			c.Args = append(c.Args, fmt.Sprintf("--max-graceful-termination-sec=%d", *options.MaxPodGracePeriod))
+func (a AutoscalerArg) Value(v interface{}) string {
+	return fmt.Sprintf("%s=%v", a.String(), v)
+}
+
+func autoscalerArgs(options *hyperv1.ClusterAutoscaling, platformType hyperv1.PlatformType, log logr.Logger) []string {
+	args := []string{}
+
+	switch options.Scaling {
+	case hyperv1.ScaleUpOnly:
+		args = append(args, ScaleDownEnabledArg.Value(false))
+	case hyperv1.ScaleUpAndScaleDown:
+		if options.ScaleDown != nil {
+			args = append(args, ScaleDownArgs(options.ScaleDown)...)
 		}
+	}
+
+	if options.MaxNodesTotal != nil {
+		args = append(args, MaxNodesTotalArg.Value(*options.MaxNodesTotal))
+	}
+
+	if options.MaxPodGracePeriod != nil {
+		args = append(args, MaxGracefulTerminationSecArg.Value(*options.MaxPodGracePeriod))
+	}
+
+	if options.MaxNodeProvisionTime != "" {
+		args = append(args, MaxNodeProvisionTimeArg.Value(options.MaxNodeProvisionTime))
+	}
+
+	if options.MaxFreeDifferenceRatioPercent != nil {
+		args = append(args, MaxFreeDifferenceRatioArg.Value(fmt.Sprintf("%.2f", float64(*options.MaxFreeDifferenceRatioPercent)/100.0)))
+	}
 
-		if options.MaxNodeProvisionTime != "" {
-			c.Args = append(c.Args, fmt.Sprintf("--max-node-provision-time=%s", options.MaxNodeProvisionTime))
+	if options.PodPriorityThreshold != nil {
+		args = append(args, ExpendablePodsPriorityCutoffArg.Value(*options.PodPriorityThreshold))
+	}
+
+	for _, ignoredLabel := range options.BalancingIgnoredLabels {
+		args = append(args, BalancingIgnoreLabelArg.Value(ignoredLabel))
+	}
+
+	if len(options.Expanders) > 0 {
+		expanders := make([]string, 0)
+		for _, v := range options.Expanders {
+			switch v {
+			case hyperv1.LeastWasteExpander:
+				expanders = append(expanders, leastWasteFlag)
+			case hyperv1.PriorityExpander:
+				expanders = append(expanders, priorityFlag)
+			case hyperv1.RandomExpander:
+				expanders = append(expanders, randomFlag)
+			default:
+				// this shouldn't happen since we have validation on the API types, but just in case
+				log.Error(errors.New("unknown priority expander"), "Unexpected Cluster Autoscaler priority expander", v)
+				continue
+			}
 		}
+		args = append(args, ExpanderArg.Value(strings.Join(expanders, ",")))
+	} else {
+		args = append(args, ExpanderArg.Value(strings.Join([]string{priorityFlag, leastWasteFlag}, ",")))
+	}
 
-		if options.PodPriorityThreshold != nil {
-			c.Args = append(c.Args, fmt.Sprintf("--expendable-pods-priority-cutoff=%d", *options.PodPriorityThreshold))
+	// Since we hardcode "balance-similar-node-groups" to true in the deployment yaml
+	// Append basic ignore labels for a specific cloud provider.
+	args = appendBasicIgnoreLabels(args, platformType)
```

Comment:
can we please handle all labels here programmatically consistently, i.e move the ones hardcoded in the manifest here